Bluetooth: tester: Fix configuration

carlescufi opened the pull request:
See commit messages

carlescufi committed:
tests: bluetooth: tester: Clean up configuration

- Use the built-in mechanism for configuration overlays
  (having a boards/ folder with <board>.conf fragments)
- Clean up variables that were giving warnings

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
tests: bluetooth: tester: Add a config for the 52840 DK

Add a configuration file for the nRF52840 DK so that it uses RTT for
debugging and enables more protection measures.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

Vudentz commented on tests/bluetooth/tester/qemu.conf:
```diff
@@ -1,49 +0,0 @@
-CONFIG_TEST=y
```
Do we really want to remove qemu support or is it because prj.conf has the same configuration?

carlescufi replied:
Correct, now `prj.conf` is concatenated with `qemu_cortex_m3.conf` automagically by the build system.

carlescufi replied:
The qemu support stays exactly the same as before.

carlescufi committed:
Bluetooth: tests: tester: Add an init log message

Add a log message to let people running the test know that the tester
booted up correctly.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>
